packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_masternode_update.test.ts

```diff
@@ -0,0 +1,979 @@
+        }
+      ]
+    }
+
+    const script = await providers.elliptic.script()
+    const txn: TransactionSegWit = await builder.masternode.update(updateMasternode, script)
+    const outs = await sendTransaction(container, txn)
+
+    const encoded: string = OP_CODES.OP_DEFI_TX_UPDATE_MASTER_NODE(updateMasternode).asBuffer().toString('hex')
+    const expectedRedeemScript = `6a${encoded}`
+
+    expect(outs.length).toStrictEqual(2)
+    expect(outs[0]).toStrictEqual({
+      n: 0,
+      tokenId: 0,
+      value: 0,
+      scriptPubKey: {
+        asm: expect.stringContaining('OP_RETURN 446654786d'),
+        hex: expectedRedeemScript,
+        type: 'nulldata'
+      }
+    })
+    expect(outs[1]).toStrictEqual({
+      n: 1,
+      tokenId: 0,
+      value: expect.any(Number),
+      scriptPubKey: {
+        addresses: [await providers.getAddress()],
+        asm: expect.any(String),
+        hex: expect.any(String),
+        reqSigs: 1,
+        type: 'witness_v0_keyhash'
+      }
+    })
+    expect(outs[1].value).toBeGreaterThan(6.99)
+    expect(outs[1].value).toBeLessThan(7)
+  })
+
+  it('should update reward address with P2WPKH address', async () => {
+    const pubKey = await providers.ellipticPair.publicKey()
+    const collateralAddress = Bech32.fromPubKey(pubKey, 'bcrt')
+    const txid = await jsonRpc.masternode.createMasternode(collateralAddress)
+    await container.generate(20)
+
+    const address = await container.getNewAddress('', 'bech32')
+    const addressDest: P2WPKH = P2WPKH.fromAddress(RegTest, address, P2WPKH)
+    const addressDestKeyHash = addressDest.pubKeyHash
+
+    const updateMasternode: UpdateMasternode = {
+      nodeId: txid,
+      updates: [
+        {
+          updateType: 0x03,
+          address: { addressType: 0x04, addressPubKeyHash: addressDestKeyHash }
+        }
+      ]
+    }
+
+    const script = await providers.elliptic.script()
+    const txn: TransactionSegWit = await builder.masternode.update(updateMasternode, script)
+    const outs = await sendTransaction(container, txn)
+
+    const encoded: string = OP_CODES.OP_DEFI_TX_UPDATE_MASTER_NODE(updateMasternode).asBuffer().toString('hex')
+    const expectedRedeemScript = `6a${encoded}`
+
+    expect(outs.length).toStrictEqual(2)
+    expect(outs[0]).toStrictEqual({
+      n: 0,
+      tokenId: 0,
+      value: 0,
+      scriptPubKey: {
+        asm: expect.stringContaining('OP_RETURN 446654786d'),
+        hex: expectedRedeemScript,
+        type: 'nulldata'
+      }
+    })
+    expect(outs[1]).toStrictEqual({
+      n: 1,
+      tokenId: 0,
+      value: expect.any(Number),
+      scriptPubKey: {
+        addresses: [await providers.getAddress()],
+        asm: expect.any(String),
+        hex: expect.any(String),
+        reqSigs: 1,
+        type: 'witness_v0_keyhash'
+      }
+    })
+    expect(outs[1].value).toBeGreaterThan(6.99)
+    expect(outs[1].value).toBeLessThan(7)
+  })
```

Comment:
I think that there should be an assertion to check that the reward address has updated correctly (for eg. by checking `listMasternodes`), and similarly for other tests

Reply:
ok will take note of this in the new PR #1894